lutris/util/display.py

```diff
@@ -130,37 +131,120 @@ def get_display_manager():
+# True if compositing has been disabled, False if not. enable_compositing
+# removes the last element, and only re-enables compositing if that element
+# was True.
+_COMPOSITING_DISABLED_STACK = []
+
+
+def _get_compositor_commands():
+    """Returns the commands to enable/disable compositing on the current
+    desktop environment as a 2-tuple.
+    """
     start_compositor = None
     stop_compositor = None
-    desktop_session = os.environ.get("DESKTOP_SESSION")
-    if desktop_session.endswith("plasma"):
-        stop_compositor = ("qdbus org.kde.KWin /Compositor org.kde.kwin.Compositing.suspend")
-        start_compositor = ("qdbus org.kde.KWin /Compositor org.kde.kwin.Compositing.resume")
-    elif (
-        desktop_session.endswith("mate")
-        and system.execute("gsettings get org.mate.Marco.general compositing-manager", shell=True) == "true"
-    ):
-        stop_compositor = ("gsettings set org.mate.Marco.general compositing-manager false")
-        start_compositor = ("gsettings set org.mate.Marco.general compositing-manager true")
-    elif (
-        desktop_session.endswith("xfce") and system.execute(
-            "xfconf-query --channel=xfwm4 --property=/general/use_compositing",
-            shell=True,
-        ) == "true"
-    ):
-        stop_compositor = ("xfconf-query --channel=xfwm4 --property=/general/use_compositing --set=false")
-        start_compositor = ("xfconf-query --channel=xfwm4 --property=/general/use_compositing --set=true")
-    elif (
-        desktop_session.endswith("deepin") and system.execute(
-            "dbus-send --session --dest=com.deepin.WMSwitcher --type=method_call "
-            "--print-reply=literal /com/deepin/WMSwitcher com.deepin.WMSwitcher.CurrentWM",
-            shell=True,
-        ) == "deepin wm"
-    ):
-        start_compositor, stop_compositor = (
-            "dbus-send --session --dest=com.deepin.WMSwitcher --type=method_call "
-            "/com/deepin/WMSwitcher com.deepin.WMSwitcher.RequestSwitchWM",
-        ) * 2
+    desktop_environment = get_desktop_environment()
+    if desktop_environment is DesktopEnvironment.PLASMA:
+        stop_compositor = ("qdbus", "org.kde.KWin", "/Compositor", "org.kde.kwin.Compositing.suspend")
+        start_compositor = ("qdbus", "org.kde.KWin", "/Compositor", "org.kde.kwin.Compositing.resume")
+    elif desktop_environment is DesktopEnvironment.MATE:
+        stop_compositor = ("gsettings", "set org.mate.Marco.general", "compositing-manager", "false")
+        start_compositor = ("gsettings", "set org.mate.Marco.general", "compositing-manager", "true")
+    elif desktop_environment is DesktopEnvironment.XFCE:
+        stop_compositor = ("xfconf-query", "--channel=xfwm4", "--property=/general/use_compositing", "--set=false")
+        start_compositor = ("xfconf-query", "--channel=xfwm4", "--property=/general/use_compositing", "--set=true")
+    elif desktop_environment is DesktopEnvironment.PLASMA:
```

Comment:
This must be a copy/paste mistake.

```suggestion
    elif desktop_environment is DesktopEnvironment.DEEPIN:
```

Reply:
Yup, thanks for catching that!